nav2_collision_monitor/src/collision_monitor_node.cpp

```diff
@@ -386,8 +386,13 @@ void CollisionMonitor::process(const Velocity & cmd_vel_in)
       break;
     }
 
-    // Update polygon coordinates
-    polygon->updatePolygon();
+    if (polygon->isUsingPolygonVelocitySelector()) {
+      // Update polygon using its velocity ranges
+      polygon->updatePolygonByVelocity(cmd_vel_in);
```

Comment:
We could just make updatePolygon take a velocity field that the normal polygon ignores that doesn't care about it. Simplifies this because we don't need to check anything